packages/protocol/test-sol/unit/common/EpochManager.t.sol

```diff
@@ -27,47 +29,52 @@ contract EpochManagerTest is Test, TestConstants, Utils08 {
   EpochRewardsMock08 epochRewards;
   ValidatorsMock08 validators;
 
-  address epochManagerInitializer;
+  address epochManagerEnabler;
   address carbonOffsettingPartner;
   address communityRewardFund;
+  address reserveAddress;
+  address scoreManagerAddress;
 
   uint256 firstEpochNumber = 100;
   uint256 firstEpochBlock = 100;
+  uint256 epochDuration = DAY;
   address[] firstElected;
 
   IRegistry registry;
-  ICeloToken celoToken;
-  CeloUnreleasedTreasure celoUnreleasedTreasure;
+  MockCeloToken08 celoToken;
+  MockCeloUnreleasedTreasure celoUnreleasedTreasure;
   ScoreManager scoreManager;
 
   uint256 celoAmountForRate = 1e24;
   uint256 stableAmountForRate = 2 * celoAmountForRate;
 
+  event EpochProcessingStarted(uint256 indexed epochNumber);
+
   function setUp() public virtual {
     epochManager = new EpochManager(true);
     sortedOracles = new MockSortedOracles();
     epochRewards = new EpochRewardsMock08();
     validators = new ValidatorsMock08();
     stableToken = new MockStableToken08();
-    celoUnreleasedTreasure = new CeloUnreleasedTreasure(false);
+    celoToken = new MockCeloToken08();
+    celoUnreleasedTreasure = new MockCeloUnreleasedTreasure();
 
     firstElected.push(actor("validator1"));
     firstElected.push(actor("validator2"));
 
-    address celoTokenAddress = actor("celoTokenAddress");
-    address scoreManagerAddress = actor("scoreManagerAddress");
-    address reserveAddress = actor("reserve");
+    scoreManagerAddress = actor("scoreManagerAddress");
+
+    reserveAddress = actor("reserve");
 
-    epochManagerInitializer = actor("initializer");
+    epochManagerEnabler = actor("epochManagerEnabler");
     carbonOffsettingPartner = actor("carbonOffsettingPartner");
     communityRewardFund = actor("communityRewardFund");
 
-    deployCodeTo("Registry.sol", abi.encode(false), REGISTRY_ADDRESS);
-    deployCodeTo("GoldToken.sol", abi.encode(false), celoTokenAddress);
+    deployCodeTo("MockRegistry.sol", abi.encode(false), REGISTRY_ADDRESS);
```

Comment:
Do we need MockRegistry? Wouldn't something like that's already on the PR?

```

function whenL2(Vm vm) public {
    vm.etch(0x4200000000000000000000000000000000000018, abi.encodePacked(bytes1(0x01)));
  }
```

Reply:
It's fine to use the above code for the L2 check because that gets deployed to the proxyAdmin address. For the case of the registry, we need the actual registry code for the tests to work. So deploying arbitrary code to the registry address will not work.